core/src/main/java/nl/nn/adapterframework/pipes/WsdlXmlValidator.java

```diff
@@ -197,8 +195,42 @@ public void configure() throws ConfigurationException {
 		super.configure();
 	}
 
+	@Override
+	protected PipeForward validate(Message messageToValidate, PipeLineSession session, boolean responseMode, String messageRoot) throws XmlValidatorException, PipeRunException, ConfigurationException {
+		String soapAction = session.get(SoapBindingConstants.SOAP_ACTION, "");
+		if(StringUtils.isNotEmpty(soapAction)) {
+			String soapBodyFromSoapAction = getSoapBodyFromSoapAction(soapAction);
+			if(StringUtils.compare(getSoapBody(), soapBodyFromSoapAction) != 0) {
+				log.debug("soapBody ["+soapBodyFromSoapAction+"] is determined from soapAction ["+soapAction+"]");
+				messageRoot = soapBodyFromSoapAction;
+			}
+		}
+		return super.validate(messageToValidate, session, responseMode, messageRoot);
+	}
+
+	private String getSoapBodyFromSoapAction(String soapAction) {
+		Map<QName, Binding> bindings = definition.getBindings();
+		for (Entry<QName, Binding> binding : bindings.entrySet()) {
+			List<BindingOperation> operations = binding.getValue().getBindingOperations();
+			for (BindingOperation bindingOperation : operations) {
+				List<ExtensibilityElement> elements = bindingOperation.getExtensibilityElements();
+				if(elements != null && !elements.isEmpty()) {
+					for (ExtensibilityElement element : elements) {
+						if(element instanceof SOAPOperation) {
+							String soapActionFromDefinition = ((SOAPOperation) element).getSoapActionURI();
+							if(soapActionFromDefinition.equals(soapAction)) {
+								return bindingOperation.getName();
```

Comment:
I think the soapBody element does not need to be equal to the name of the bindingOperation, is it?

Reply:
True, the referenced element in input/output definition should be used